src/GraphQL/Utilities/SchemaPrinter.cs

```diff
@@ -322,9 +329,36 @@ public string FormatDefaultValue(object value)
                 return value.ToString().ToLower();
             }
 
+            if (IsEnumType(graphType))
+            {
+                return "{0}".ToFormat(SerializeEnumValue(graphType, value));
+            }
+
             return "{0}".ToFormat(value);
         }
 
+        private static bool IsEnumType(IGraphType type)
+        {
+            if (type is NonNullGraphType)
+            {
+                var nullable = (NonNullGraphType)type;
+                type = nullable.ResolvedType;
+            }
+
+            return type is EnumerationGraphType;
+        }
+
+        private static object SerializeEnumValue(IGraphType type, object value)
+        {
+            if (type is NonNullGraphType)
+            {
+                var nullable = (NonNullGraphType)type;
+                type = nullable.ResolvedType;
+            }
+
+            return ((EnumerationGraphType)type).Serialize(value);
+        }
```

Comment:
Do we not have to worry about printing the `!` for explicitly non-null enum values?  Seems like we should?  Something like the `ResolveName` function in this same file?

Reply:
This is for the enum _value_, not the type. Essentially, if you have a resolver: `foo(arg: SomeEnum = X)`, it ensures that `X` is printed as the name of the enum value, i.e., the serialised form of value `X` in type `SomeEnum`.

In other words: If `SomeEnum` is defined for values `FooEnum.Foo` with name = `FOO` and `FooEum.Bar` with name `BAR` (using `FooEnum` here just to make it clear that the underlying linked value of an enum doesn't necessarily have to match with the GraphQL enum type (name of type and values)), this would print `FOO` instead of `Foo` if the value `X` was `FooEnum.Foo`. Make sense?